intrinsics/fnsub.go

```diff
@@ -26,6 +32,23 @@ func FnSub(name string, input interface{}, template interface{}) interface{} {
 				return src
 			}
 		}
+
+	case string:
+		// Look up references for each of the variables
+		regex := regexp.MustCompile(`\$\{([0-9A-Za-z]+)\}`)
+		variables := regex.FindAllStringSubmatch(val, -1)
+		for _, variable := range variables {
+			resolved := Ref("Ref", variable[1], template)
+			if resolved != nil {
+				if replacement, ok := resolved.(string); ok {
+					val = strings.Replace(val, variable[0], replacement, -1)
+				}
+			} else {
+				// The reference couldn't be resolved, so just strip the variable
+				val = strings.Replace(val, variable[0], "", -1)
```

Comment:
What is variable[0] and variable[1]? Can you assign them to a separate variable for clarity?